include/revng/TupleTree/TupleTreeDiff.h

```diff
@@ -74,7 +129,113 @@ struct TupleTreeDiff {
     dump(OutputStream);
   }
 
-  void apply(T &M) const;
+  void apply(TupleTree<T> &M) const;
+};
+
+template<typename T>
+concept DiffSpecialization = is_specialization_v<T, TupleTreeDiff>;
+
+template<typename T>
+concept ChangeSpecialization = is_specialization_v<T, Change>;
+
+template<DiffSpecialization T>
+struct llvm::yaml::MappingTraits<T> {
+  static void mapping(IO &io, T &info) {
+    io.mapOptional("changes", info.Changes);
```

Comment:
In every `YAMLTraits` we use the same casing as the type.
We might wanna change, but let's be coherent.